protostar/cairo_testing/cairo_test_execution_state.py

```diff
@@ -44,4 +48,5 @@ def fork(self) -> Self:
             output_recorder=self.output_recorder.fork(),
             stopwatch=self.stopwatch.fork(),
             cheatable_starknet_facade=self.cheatable_starknet_facade.fork(),
+            event_expectations=self.event_expectations.copy(),
```

Comment:
Theoretically, someone could use expect events in "setup", if we plan to make this hint available there.

Reply:
I don't think we should allow that TBH, even if we would, i think the lifespan of the expectation should be the scope of the test/setup function. Otherwise it would make some bizzare scenarios. 

It's up for discussion though.

Reply:
Transferring this discussion to private. Will merge as-is, we can follow-up later, this PR has been up for long enough, and i figure the code is backwards-compatible as-is